src/Cli/func/Common/ProcessManager.cs

```diff
@@ -26,6 +26,11 @@ public IEnumerable<IProcessInfo> GetProcessesByName(string processName)
                 .Select(p => new ProcessInfo(p));
         }
 
+        public void KillMainProcess()
```

Comment:
Still not a fan of this as an API on `IProcessManager`.

Reply:
Can you share more? I can just call `Process.GetCurrentProcess().Kill();` directly without it being in this interface, but would like to understand why.

Context: I originally added it to the interface here because I was using the process manager within the CancelKeyHandler, but the code has been refactored since, leaving the control with the registered actions instead. I was also mocking the behaviour for testing which is why I wanted it to be part of an interface.

Reply:
`IProcessManager` does not manage the lifetime of the main process. I looked for this being mocked in tests, which would have helped better understand the motivation, but didn't see anything leveraging this.

Ultimately, it comes down to the role of `IProcessManager`: This is an abstraction for the service responsible for managing child processes launched by Core Tools, not for the main application lifecycle/process. Looks like there is some general process logic that was already bleeding into that interface, but we shouldn't expand on that.

Reply:
> but didn't see anything leveraging this.

must have also been removed with the refactor - I can revert this change